src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
```

Comment:
That ordering problem is why I probed about the change in ordering re: when `WT_CONN_CLOSING` is set in `__wt_connection_close`. I'll open a new ticket requesting a safety mechanism around shutdown of worker threads.